File: BA-Guest/customCell/communitycell.m
//
//  communitycell.m
//  BA-Guest
//
//  Created by roberto ramirez on 2/4/14.
//  Copyright (c) 2014 lovetthomes. All rights reserved.
//

#import "communitycell.h"

@implementation communitycell

@synthesize idcia, cianame, city, webarea, webnm;

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        // Initialization code
    }
    return self;
}

- (void)drawRect:(CGRect)rect{
    int tx=2;
    int dw =self.frame.size.width;
    UILabel* lblname = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, 110, 44)];
    lblname.text=idcia;
    //    lblname.autoresizingMask=UIViewAutoresizingFlexibleWidth;
    [self.contentView addSubview: lblname];
    tx=tx +110;
    
    UILabel * label = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, 1, 44)];
    label.backgroundColor=[UIColor colorWithRed:0.88 green:0.88 blue:0.88 alpha:1];
    //    label.autoresizingMask=UIViewAutoresizingFlexibleLeftMargin;
    [self.contentView addSubview: label];
    
    label = [[UILabel alloc] initWithFrame: CGRectMake( 0, 43, dw, 1)];
    label.backgroundColor=[UIColor colorWithRed:0.88 green:0.88 blue:0.88 alpha:1];
    //    label.autoresizingMask=UIViewAutoresizingFlexibleLeftMargin;
    [self.contentView addSubview: label];
    
    tx = tx+3;
    UILabel* lblpname = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, dw*.2-2, 44)];
    lblpname.text=cianame;
    //    lblpname.autoresizingMask=UIViewAutoresizingFlexibleWidth;
    [self.contentView addSubview: lblpname];
    tx=tx +dw*.2-2;
    
    label = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, 1, 44)];
    label.backgroundColor=[UIColor colorWithRed:0.88 green:0.88 blue:0.88 alpha:1];
    //    label.autoresizingMask=UIViewAutoresizingFlexibleLeftMargin;
    [self.contentView addSubview: label];
    tx=tx+3;
    
    lblpname = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, dw*.12-2, 44)];
    lblpname.text=city;
    //    lblpname.autoresizingMask=UIViewAutoresizingFlexibleWidth;
    [self.contentView addSubview: lblpname];
    tx=tx +dw*.12-2;
    
    label = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, 1, 44)];
    label.backgroundColor=[UIColor colorWithRed:0.88 green:0.88 blue:0.88 alpha:1];
    //    label.autoresizingMask=UIViewAutoresizingFlexibleLeftMargin;
    [self.contentView addSubview: label];
    tx=tx+3;
    
    lblpname = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, dw*.24-2, 44)];
    lblpname.text=webarea;
    //    lblpname.autoresizingMask=UIViewAutoresizingFlexibleWidth;
    [self.contentView addSubview: lblpname];
    
    tx=tx +dw*.24-2;
    
    label = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, 1, 44)];
    label.backgroundColor=[UIColor colorWithRed:0.88 green:0.88 blue:0.88 alpha:1];
    //    label.autoresizingMask=UIViewAutoresizingFlexibleLeftMargin;
    [self.contentView addSubview: label];
    tx=tx+3;
    
    lblpname = [[UILabel alloc] initWithFrame: CGRectMake( tx, 0, dw-tx, 44)];
    lblpname.text=webnm;
    //    lblpname.autoresizingMask=UIViewAutoresizingFlexibleWidth;
    [self.contentView addSubview: lblpname];

    
}


- (void)setSelected:(BOOL)selected animated:(BOOL)animated
{
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: BA-Guest/views/communityCell.swift
//
//  communityCell.swift
//  BA-Guest
//
//  Created by April on 3/4/16.
//  Copyright © 2016 lovetthomes. All rights reserved.
//

import UIKit

@objcMembers class communityCell: UITableViewCell {

    @IBOutlet var ciaName: UILabel!
    @IBOutlet var cityName: UILabel!
    @IBOutlet var areaName: UILabel!
    @IBOutlet var communityName: UILabel!
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        for a in self.contentView.subviews {
            if a.frame.width == 1 {
                for b in a.constraints{
                    b.constant = 1.0 / UIScreen.main.scale
                }
            }
        }
        
        let v = UIView()
        self.contentView.addSubview(v)
        v.backgroundColor = UIColor(red: 204/255.0, green: 204/255.0, blue: 204/255.0, alpha: 1)
        let leadingConstraint = NSLayoutConstraint(item:v,
            attribute: .leadingMargin,
            relatedBy: .equal,
            toItem: self.contentView,
            attribute: .leadingMargin,
            multiplier: 1.0,
            constant: 0);
        let trailingConstraint = NSLayoutConstraint(item:v,
            attribute: .trailingMargin,
            relatedBy: .equal,
            toItem: self.contentView,
            attribute: .trailingMargin,
            multiplier: 1.0,
            constant: 0);
        
        let bottomConstraint = NSLayoutConstraint(item: v,
            attribute: .bottomMargin,
            relatedBy: .equal,
            toItem: self.contentView,
            attribute: .bottomMargin,
            multiplier: 1.0,
            constant: 0);
        
        let heightContraint = NSLayoutConstraint(item: v,
            attribute: .height,
            relatedBy: .equal,
            toItem: nil,
            attribute: .notAnAttribute,
            multiplier: 1.0,
            constant: 1.0 / (UIScreen.main.scale));
        v.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([leadingConstraint, trailingConstraint, bottomConstraint, heightContraint])
    }
    public func setCellContent(_ item : wcfCommunityItem){
        ciaName.text = item.ciaName
        cityName.text = item.city
        areaName.text = item.webArea
        communityName.text = item.webCommunity
    }
    
    override func setHighlighted(_ highlighted: Bool, animated: Bool) {
        if highlighted {
            self.contentView.backgroundColor = CConstants.SearchBarBackColor
        }else{
            self.contentView.backgroundColor = UIColor.white
        }
    }
    
    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        if selected {
            self.contentView.backgroundColor = CConstants.SearchBarBackColor
        }else{
            self.contentView.backgroundColor = UIColor.white
        }
    }
    
}
